File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        part = []
        res = []
        def backtrack(nums):
            if not nums:
                res.append(part[:])
                return 
            part.append(nums[0])
            backtrack(nums[1:])
            part.pop()
            backtrack(nums[1:])
        backtrack(nums)
        return res


        

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        backtrack(0,nums);
    
        return res;
    }

private:
    vector<vector<int>> res;
    vector<int> part;
    void backtrack(int cur,vector<int>& nums){
        if (cur==nums.size()){
            res.push_back(part);
            return;
        }
        part.push_back(nums[cur]);
        backtrack(cur+1,nums);
        part.pop_back();
        backtrack(cur+1,nums);
    }
};